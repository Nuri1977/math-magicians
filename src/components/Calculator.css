.calc-container {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-auto-rows: 40px;
  font-size: 1rem;
  font-weight: bold;
  background-color: gray;
  max-width: 240px;
}

.span-four {
  grid-column: 1 / -1;
  text-align: right;
  padding-right: 5px;
  padding-top: 12px;
  color: antiquewhite;
}

.span-two {
  grid-column: 1 / 3;
}

.orange {
  background: orangered;
  border: 1px solid grey;
}
