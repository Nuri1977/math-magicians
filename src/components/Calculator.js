import './Calculator.css';
import { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [total, setTotal] = useState('0');
  const [next, setNext] = useState('');
  const [operation, setOperation] = useState('');

  const handleClick = (event) => {
    const value = calculate({ total, operation, next }, event.target.innerText);

    if (value.total != null) {
      setTotal(value.total);
    }
    setNext(value.next);

    if (value.operation != null) {
      setOperation(value.operation);
    }
  };

  return (
    <div className="container flex align-start">
      <div className="calc-title">
        <h1>Let&apos;s do some math!</h1>
      </div>
      <div className="calc-container">
        <div className="span-four">
          { total }
          { operation }
          { next }
        </div>
        <button type="button" onClick={handleClick}>AC</button>
        <button type="button" onClick={handleClick}>+/-</button>
        <button type="button" onClick={handleClick}>%</button>
        <button type="button" onClick={handleClick} className="orange">รท</button>
        <button type="button" onClick={handleClick}>7</button>
        <button type="button" onClick={handleClick}>8</button>
        <button type="button" onClick={handleClick}>9</button>
        <button type="button" onClick={handleClick} className="orange">x</button>
        <button type="button" onClick={handleClick}>4</button>
        <button type="button" onClick={handleClick}>5</button>
        <button type="button" onClick={handleClick}>6</button>
        <button type="button" onClick={handleClick} className="orange">-</button>
        <button type="button" onClick={handleClick}>1</button>
        <button type="button" onClick={handleClick}>2</button>
        <button type="button" onClick={handleClick}>3</button>
        <button type="button" onClick={handleClick} className="orange">+</button>
        <button type="button" onClick={handleClick} className="span-two">0</button>
        <button type="button" onClick={handleClick}>.</button>
        <button type="button" onClick={handleClick} className="orange">=</button>
      </div>
    </div>
  );
};

export default Calculator;
