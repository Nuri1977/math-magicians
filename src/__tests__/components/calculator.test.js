import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Calculator from '../../components/Calculator';

describe('Testing Calculator component', () => {
  it('Calculatore renders correctly', () => {
    const tree = renderer
      .create(<Calculator />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Testing Calculator buttons', () => {
  it('Calculator AC button ', () => {
    render(<Calculator />);
    const buttonElement = screen.getByText('AC');
    expect(buttonElement).toBeInTheDocument();
  });
  it('Calculator 1 button', () => {
    render(<Calculator />);
    const buttonElement = screen.getByText('1');
    expect(buttonElement).toBeEnabled();
  });
  it('Calculator  button clicked', () => {
    render(<Calculator />);
    const buttonElement = screen.getByRole('button', { name: /2/i });
    expect(buttonElement).toBeVisible();
  });
});
