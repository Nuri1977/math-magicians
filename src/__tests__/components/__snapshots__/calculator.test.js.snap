// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Calculator component Calculatore renders correctly 1`] = `
<div
  className="container flex align-start"
>
  <div
    className="calc-title"
  >
    <h1>
      Let's do some math!
    </h1>
  </div>
  <div
    className="calc-container"
  >
    <div
      className="span-four"
      data-testid="value"
    >
      0
      
      
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="span-two"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="orange"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
